integration-test/test-fingerprint.js

```diff
@@ -17,170 +19,118 @@ const expectedFingerprintValues = {
-            const result = await runTest(testCase)
-
-            const result2 = await runTest(testCase)
-            expect(result.canvas).toEqual(result2.canvas)
-            expect(result.plugin).toEqual(result2.plugin)
-        })
-    }
-
-    it('Fingerprints should not match across first parties', async () => {
-        const canvas = new Set()
-        const plugin = new Set()
 
         for (const testCase of tests) {
-            const result = await runTest(testCase)
+            test(`Fingerprints should not change amongst page loads test ${testCase.url}`, async ({ page, altServerPort }) => {
+                console.log('running:', altServerPort)
+                const result = await runTest(page, testCase)
 
-            // Add the fingerprints to a set, if the result doesn't match it won't be added
-            canvas.add(JSON.stringify(result.canvas))
-            plugin.add(JSON.stringify(result.plugin))
+                const result2 = await runTest(page, testCase)
+                expect(result.canvas).toEqual(result2.canvas)
+                expect(result.plugin).toEqual(result2.plugin)
+            })
         }
 
-        // Ensure that the number of test pages match the number in the set
-        expect(canvas.size).toEqual(tests.length)
-        expect(plugin.size).toEqual(1)
-    })
-})
+        test('Fingerprints should not match across first parties', async ({ page, altServerPort }) => {
+            console.log('running:', altServerPort)
+            const canvas = new Set()
+            const plugin = new Set()
 
-describe('Verify injected script is not visible to the page', () => {
-    let browser
-    let teardown
-    let setupServer
-    beforeAll(async () => {
-        ({ browser, setupServer, teardown } = await setup())
+            for (const testCase of tests) {
+                const result = await runTest(page, testCase)
 
-        setupServer('8080')
-        setupServer('8383')
-    })
-    afterAll(async () => {
-        await teardown()
+                // Add the fingerprints to a set, if the result doesn't match it won't be added
+                canvas.add(JSON.stringify(result.canvas))
+                plugin.add(JSON.stringify(result.plugin))
+            }
+
+            // Ensure that the number of test pages match the number in the set
+            expect(canvas.size).toEqual(tests.length)
+            expect(plugin.size).toEqual(1)
+        })
     })
 
-    tests.forEach(test => {
-        it('Fingerprints should not match across first parties', async () => {
-            const page = await browser.newPage(pagePath)
+    test.describe.serial('Verify injected script is not visible to the page', () => {
+        tests.forEach(testCase => {
+            test(`Fingerprints should not match across first parties ${testCase.url}`, async ({ page, altServerPort }) => {
+                console.log('running:', altServerPort)
```

Comment:
these comments are to keep typescript/eslint happy

Reply:
We could just drop the port usage in the definition instead?

Reply:
no because the `altServerPort` variable is only available inside the test. Could just eslint-disable this warning if we prefer.

Reply:
Maybe I'm misunderstanding...

But we're destructuring a property? Are you saying without this we'd not trigger the altServerPort extension?
I'll not block on this if that's the case but it's a weird side effect, can it not be a method that's triggered before the FP tests?

Reply:
I'm trying to follow docs/best-practices so that we can point people to official sources where it makes sense. In that context, I used https://playwright.dev/docs/test-fixtures to create & tear down the server on demand, per test. The docs there give the [reasons for it](https://playwright.dev/docs/test-fixtures#with-fixtures).

Actually this is exactly how `page` works too, so technically whilst it's a little odd for us (coming from puppeteer) it's the 'blessed' way I believe.

Having said that, I also understand the weirdness. In this case we just need the argument to be there for the side effects to run and be cleaned up, but because the test URL is hard-coded above in this file, we don't actually end up using it 😂 

This was all in the idea of not changing too much, maybe we should?

Reply:
Makes sense, it's a bit weird but I can live with it.

Maybe make a task to discuss?

- as server.mjs isn't reused elsewhere so we have a bunch of code remapping code into altServerPort methods and could be unified.
- In that work we could refactor to a method that accepts a port/filepath as you say.